plugins/main/public/components/endpoints-summary/agent/index.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useEffect } from 'react';
+import { EuiPage, EuiPageBody, EuiProgress } from '@elastic/eui';
+import { getErrorOrchestrator } from '../../../react-services/common-services';
+import { UI_LOGGER_LEVELS } from '../../../../common/constants';
+import { UI_ERROR_SEVERITIES } from '../../../react-services/error-orchestrator/types';
+import { AgentsWelcome } from '../../common/welcome/agents-welcome';
+import { Agent } from '../types';
+import {
+  getAngularModule,
+  getCore,
+  getDataPlugin,
+} from '../../../kibana-services';
+import { MainSyscollector } from '../../agents/syscollector/main';
+import { MainAgentStats } from '../../agents/stats';
+import WzManagementConfiguration from '../../../controllers/management/components/management/configuration/configuration-main.js';
+import { getAgentsService } from '../services';
+import { ShareAgent } from '../../../factories/share-agent';
+import store from '../../../redux/store';
+import { updateCurrentAgentData } from '../../../redux/actions/appStateActions';
+import { endpointSummary } from '../../../utils/applications';
+
+export const AgentView = () => {
+  const $injector = getAngularModule().$injector;
+
+  const $router = $injector.get('$route');
+  const $commonData = $injector.get('commonData');
+  const $reportingService = $injector.get('reportingService');
+
+  const shareAgent = new ShareAgent();
+
+  const savedTimefilter = $commonData.getTimefilter();
+  if (savedTimefilter) {
+    getDataPlugin().query.timefilter.timefilter.setTime(savedTimefilter);
```

Comment:
**nitpick**: here we are using the global timefilter, and maybe we should use the data source interface instead. This behavior could not be supported, but we should add it to avoid the interaction with the global filter manager directly.

Reply:
The useDataSource and useSearchBar hooks will be implemented when replacing the Datepicker with the SearchBar component in a future iteration. This PR is focused on removing the agent controller.